browser/ui/webui/brave_new_tab_message_handler.cc

```diff
@@ -340,6 +346,21 @@ void BraveNewTabMessageHandler::HandleSaveNewTabPagePref(
   // Collect args
   std::string settingsKeyInput = args->GetList()[0].GetString();
   auto settingsValue = args->GetList()[1].Clone();
+  std::string settingsKey;
+
+  // specifically handle string types first
+  if (settingsValue.is_string()) {
+    const auto settingsValueString = settingsValue.GetString();
+    if (settingsKeyInput == "clockFormat") {
+      settingsKey = kNewTabPageClockFormat;
+    } else {
+      LOG(ERROR) << "Invalid setting key";
+      return;
+    }
+    prefs->SetString(settingsKey, settingsValueString);
+    return;
+  }
+
   // Validate args
   // Note: if we introduce any non-bool settings values
```

Comment:
nit: maybe change this comment, since now we have introduced non-bool settings values 😉

Reply:
done